src/main/java/org/folio/tlib/postgres/impl/PgCqlQueryImpl.java

```diff
@@ -0,0 +1,270 @@
+      }
+      if (c == '\\') {
+        backslash = true;
+      } else {
+        backslash = false;
+        pgTerm.append(c);
+      }
+    }
+    if (backslash) {
+      pgTerm.append('\\');
+    }
+    if (!ft) {
+      return field.getColumn() + " " + basicOp(termNode) + " E'" + pgTerm + "'";
+    }
+    return "to_tsvector('" + language + "', " + field.getColumn() + ") @@ plainto_tsquery('"
+        + language + "', E'" + pgTerm + "')";
+  }
+
+  String handleTypeNumber(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String cqlTerm = termNode.getTerm();
+    for (int i = 0; i < cqlTerm.length(); i++) {
+      char c = cqlTerm.charAt(i);
+      switch (c) {
+        case '.':
+        case ',':
+        case '+':
+        case '-':
+          break;
+        default:
+          if (!Character.isDigit(c)) {
+            throw new IllegalArgumentException("Bad numeric for: " + termNode.toCQL());
+          }
+      }
+    }
+    return field.getColumn() + numberOp(termNode) + cqlTerm;
+  }
+
+  String handleTypeBoolean(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String cqlTerm = termNode.getTerm();
+    String pgTerm;
+    if ("false".equalsIgnoreCase(cqlTerm)) {
+      pgTerm = "FALSE";
+    } else if ("true".equalsIgnoreCase(cqlTerm)) {
+      pgTerm = "TRUE";
+    } else {
+      throw new IllegalArgumentException("Bad boolean for: " + termNode.toCQL());
+    }
+    return field.getColumn() + basicOp(termNode) + pgTerm;
+  }
+
+  String whereHandle(CQLNode node) {
+    if (node == null) {
+      return null;
+    }
+    if (node instanceof CQLBooleanNode) {
+      CQLBooleanNode booleanNode = (CQLBooleanNode) node;
+      String left = whereHandle(booleanNode.getLeftOperand());
+      String right = whereHandle(booleanNode.getRightOperand());
+      switch (booleanNode.getOperator()) {
+        case OR:
+          if (right != null && left != null) {
+            return "(" + left + " OR " + right + ")";
+          }
+          return null;
+        case AND:
+          if (right != null && left != null) {
+            return "(" + left + " AND " + right + ")";
+          } else if (right != null) {
+            return right;
+          } else {
+            return left;
+          }
+        case NOT:
+          if (right != null && left != null) {
+            return "(" + left + " AND NOT " + right + ")";
+          } else if (right != null) {
+            return "NOT (" + right + ")";
+          }
+          return "FALSE";
+        default:
+          throw new IllegalArgumentException("Unsupported operator "
+              + booleanNode.getOperator().name());
+      }
+    } else if (node instanceof CQLTermNode) {
+      CQLTermNode termNode = (CQLTermNode) node;
+      PgCqlField field = fields.get(termNode.getIndex().toLowerCase());
```

Comment:
Ah, I guess that `dc.Title` below is testing this `toLowerCase()`. Makes sense.

Reply:
Right